OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -434,5 +430,33 @@ bool ShowPaletteDropdown(DropDownButtonWidget dropdown, World world)
 			dropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 280, palettes, setupItem);
 			return true;
 		}
+
+		string GetSourceDisplayName(IReadOnlyPackage source)
+		{
+			if (source == null)
+				return "All Packages";
+
+			// Packages that are explicitly mounted in the filesystem use their explicit mount name
+			var fs = (OpenRA.FileSystem.FileSystem)modData.DefaultFileSystem;
+			var name = fs.GetPrefix(source);
+
+			// Fall back to the path relative to the mod, engine, or support dir
+			if (name == null)
+			{
+				name = source.Name;
+				var compare = Platform.CurrentPlatform == PlatformType.Windows ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
+				if (name.StartsWith(modData.Manifest.Package.Name, compare))
+					name = "${0}".F(modData.Manifest.Id) + name.Substring(modData.Manifest.Package.Name.Length);
```

Comment:
It feels a bit weird to use the `.F` syntax only for one case and `+` in the rest of the file.